Cabal/Distribution/Simple/SrcDist.hs

```diff
@@ -150,55 +163,55 @@ listPackageSourcesOrdinary :: Verbosity
                            -> [PPSuffixHandler]
                            -> IO [FilePath]
 listPackageSourcesOrdinary verbosity pkg_descr pps =
+-- TODO: We should remove the verbosity parameter.
+--       Function defaultPackageDesc isn't using it internally.
   fmap concat . sequenceA $
   [
     -- Library sources.
     fmap concat
-    . withAllLib $ \Library { exposedModules = modules, libBuildInfo = libBi } ->
-     allSourcesBuildInfo libBi pps modules
+    . withAllLib $ \Library {exposedModules = ex, libBuildInfo = bi} -> do
+       let exposed = filterAutogenModules pkg_descr bi ex
+       let others = otherModules $ filterAutogenBuildInfo pkg_descr bi
```

Comment:
Yes, at that time it was in discussion where to put the autogen field (really just me trying that the field was global)